src/pal/src/arch/arm/callsignalhandlerwrapper.S

```diff
@@ -18,13 +18,30 @@ C_FUNC(SignalHandlerWorkerReturnOffset\Alignment):
 // address set to SignalHandlerWorkerReturn during SIGSEGV handling.
 // It enables the unwinder to unwind stack from the handling code to the actual failure site.
 NESTED_ENTRY CallSignalHandlerWrapper\Alignment, _TEXT, NoHandler
-    alloc_stack     (8 + \Alignment) // red zone + alignment
-    PROLOG_PUSH  "{r7, lr}"
+#ifndef __linux__
+__StackAllocationSize = (8 + 4 + \Alignment) // red zone + alignment
+    alloc_stack __StackAllocationSize
+    PROLOG_PUSH "{r7, r11, lr}"
     bl      EXTERNAL_C_FUNC(signal_handler_worker)
 LOCAL_LABEL(SignalHandlerWorkerReturn\Alignment):
-    EPILOG_POP "{r7, lr}"
-    free_stack (8 + \Alignment)    
+    EPILOG_POP "{r7, r11, lr}"
+    // We don't use free_stack here because it lead to compiler generates
```

Comment:
Actually, I've realized that the `free_stack` macro should be fixed to not to contain the `.pad`, since the EPILOG_POP also doesn't modify the frame offset in any way. Could you please make that change and put back the free_stack here?

Reply:
Done.